at_onboarding_flutter/lib/screen/at_onboarding_activate_screen.dart

```diff
@@ -0,0 +1,212 @@
+            constraints: const BoxConstraints(
+              maxWidth: 400,
+            ),
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              mainAxisAlignment: MainAxisAlignment.center,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: const [
+                AtSyncIndicator(),
+                SizedBox(height: 10),
+                Text('Please wait while fetching atSign status'),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  //It will validate the person with atsign, email and the OTP.
+  //If the validation is successful, it will return a cram secret for the user to login
+  void loginWithAtsignAfterReset(BuildContext context) async {
+    String? atsign = widget.atSign ?? _onboardingService.currentAtsign;
+    atsign ??= await _onboardingService.getAtSign();
+    if (atsign != null) {
+      atsign = atsign.split('@').last;
+    }
+    dynamic data;
+
+    dynamic response = await _freeAtsignService
+        .loginWithAtsign(atsign ?? (widget.atSign ?? ''));
+    if (response.statusCode == 200) {
+      data = response.body;
+      data = jsonDecode(data);
+
+      final result = await AtOnboardingOTPScreen.push(
+        context: context,
+        atSign: atsign ?? (widget.atSign ?? ''),
+        hideReferences: false,
+      );
+
+      if (result != null) {
+        String? secret = result.secret?.split(':').last ?? '';
+        _processSharedSecret(atsign: result.atSign, secret: secret);
+      } else {
+        if (!mounted) return;
+        Navigator.pop(context, AtOnboardingResult.cancelled());
+      }
+    } else {
+      data = response.body;
+      data = jsonDecode(data);
+      String errorMessage = data['message'];
+      await showErrorDialog(errorMessage);
+    }
+  }
+
+  Future<void> showErrorDialog(String? errorMessage) async {
+    return AtOnboardingDialog.showError(
+      context: context,
+      message: errorMessage ?? '',
+      onCancel: () {
+        Navigator.of(context).pop();
+      },
+    );
+  }
+
+  void _showReferenceWebview() {
+    if (Platform.isAndroid || Platform.isIOS) {
+      AtOnboardingReferenceScreen.push(
+        context: context,
+        title: AtOnboardingStrings.faqTitle,
+        url: AtOnboardingStrings.faqUrl,
+      );
+    } else {
+      launchUrl(
+        Uri.parse(
+          AtOnboardingStrings.faqUrl,
+        ),
+      );
+    }
+  }
+
+  Future<dynamic> _processSharedSecret({
+    required String atsign,
+    required String secret,
+  }) async {
+    dynamic authResponse;
+    try {
+      atsign = atsign.startsWith('@') ? atsign : '@$atsign';
+
+      bool isExist = await _onboardingService.isExistingAtsign(atsign);
+      if (isExist) {
+        await _showAlertDialog(
+            AtOnboardingErrorToString().pairedAtsign(atsign));
+        return;
+      }
+      await Future.delayed(const Duration(seconds: 10));
```

Comment:
@sachins-geekyants @sonle-geekyants Can we add the reason of adding this delay here.

Reply:
Added comment: 
//Delay 10s for waiting for ServerStatus change to teapot when activating an atsign

Sachin also talks about this issue. Don't have any else solution for now.